ulist/ulist.go

```diff
@@ -0,0 +1,164 @@
+package ulist
+
+import (
+	"github.com/zyedidia/generic/list"
+)
+
+//          ---------    ---------
+//  UList:  | Block | <-> | Block | <-> ...
+//          ---------    ---------
+// A UList is represented internally as a list.List (doubly linked list)
+// of pointers to blocks of entries.
+//
+// Block:  []V{ ... }
+// A Block is a slice of entries and forms the Node in the list.List.
+
+// Type alias for a pointer to a block of entries.
+type blockPtr[V any] *[]V
```

Comment:
Is it necessary to make this a `*[]V`, or could it just be a `[]V` to avoid one additional layer of indirection.

Reply:
Good idea, made the change and pushed [eed94a9](https://github.com/zyedidia/generic/pull/46/commits/eed94a920ad25cfaf20b08623d951c3e405b54b5)